build_functions/CMakeLists.txt

```diff
@@ -193,3 +193,22 @@ function(compileTargetAsC11 theTarget)
     set_target_properties(${theTarget} PROPERTIES CXX_STANDARD 11)
   endif()
 endfunction()
+
+set(LIBS_TO_BE_REAL_CHECKED CACHE INTERNAL "LIBS_TO_BE_CHECKED")
+
+function (add_library())
+    _add_library(${ARGV})
+    get_target_property(lib_type ${ARGV0} TYPE)
+    if( (NOT ${CMAKE_CURRENT_LIST_DIR} MATCHES "deps") AND # ignore dependency libs
+        (NOT ${ARGV0} MATCHES "_ut_lib|int_lib$") AND # ignore test libs
+        (NOT ${lib_type} STREQUAL "INTERFACE_LIBRARY")) # ignore interface libs
+        set(LIBS_TO_BE_REAL_CHECKED "${LIBS_TO_BE_REAL_CHECKED};${ARGV0}" CACHE INTERNAL "LIBS_TO_BE_CHECKED")
+    endif()
+endfunction()
+
+function (get_target_files())
+    list(SUBLIST ARGV 1 -1 libraries)
```

Comment:
> list(SUBLIST ARGV 1 -1 libraries) [](http://example.com/codeflow?start=3&length=34)

Why use here list/sublist and then foreach. I mean, what happens if 
```
foreach(lib ${ARGV}) 
```
?

What's the meaning of ARGV0 here and why is it skipped by sublist?

Reply:
The first argument (ARGV0) to the function is a variable that will contain the generator expressions for the paths. The rest of the arguments (list(SUBLIST ARGV 1 -1 libraries)) are the libs for which the paths must be found.